### PR TITLE
<OOP school library: set up associations>

### PR DESCRIPTION
**Pull request by Bestbynature**
## :zap: Features included in this pull request are as follows ✔️ :

- [x] A new class called `Classroom` was created ☘️

- [x] The has-many/belongs-to relationship between Classroom and Student was implemented ☘️

- [x] A new class `Book` was created with @title and @author instance variables initialized in the constructor ☘️

- [x] A class `Rental` was created with the @date instance variable initialized in the constructor ☘️

- [x] The many-to-many (also has-many-through) relationship between Person and Book using the intermediate class `Rental` was implemented ☘️

### CODE REVIEW

**Review by NahnahAJ — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @Bestbynature , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- Linters well setup and passing :heavy_check_mark: 
- Good use of GitFlow :heavy_check_mark: 
- Professional README :heavy_check_mark: 
- Code is clean and well-structured :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 1]
```diff
@@ -1,18 +1,25 @@
 require_relative 'person'
```
[the remark this answers is not in the document]
Comment (Bestbynature) [reply student.rb:1]: @NahnahAJ Thank you so much for your input. Please about this last point, is this method implementation correct for adding the rental?


def add_rental(rental)
    @rentals << rental
  end

[the remark this answers is not in the document]
Comment (Bestbynature) [reply student.rb:1]: Also, Please clarify. Am I not repeating setters for rentals if I create a method add_rental again after I had already used attr_accessor :rentals

[the remark this answers is not in the document]
Comment (NahnahAJ) [reply student.rb:1]: Hello @Bestbynature That method should look something like this:

```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  attr_accessor :date
+  attr_reader :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+
+    person.add_rental(self)
+    book.add_rental(self)
```
[the remark this answers is not in the document]
Comment (Bestbynature) [reply rental.rb:11]: Thank you so much

File: student.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bestbynature) [reply student.rb:1]: Thank you so much

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Bestbynature, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**